fl4health/clients/basic_client.py

```diff
@@ -394,38 +626,79 @@ def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
         """
         User defined method that returns a PyTorch Train DataLoader
         and a PyTorch Validation DataLoader
+
+        Args:
+            config (Config): The config from the server.
+
+        Returns:
+            Tuple[DataLoader, DataLoader]: The client train and validation loader.
+
+        Raises:
+            NotImplementedError: To be defined in child class.
         """
         raise NotImplementedError
 
     def get_criterion(self, config: Config) -> _Loss:
         """
         User defined method that returns PyTorch loss to train model.
+
+        Args:
+            config (Config): The config from the server.
+
+        Raises:
+            NotImplementedError: To be defined in child class.
         """
         raise NotImplementedError
 
     def get_optimizer(self, config: Config) -> Union[Optimizer, Dict[str, Optimizer]]:
         """
         Method to be defined by user that returns the PyTorch optimizer used to train models locally
         Return value can be a single torch optimizer or a dictionary of string and torch optimizer.
+
+        Args:
+            config (Config): The config sent from the server.
+
+        Returns:
+            Union[Optimizer, Dict[str, Optimizer]]: An optimizer or dictionary of optimizers to
+            train model.
+
+        Raises:
+            NotImplementedError: To be defined in child class.
         """
         raise NotImplementedError
 
     def get_model(self, config: Config) -> nn.Module:
         """
-        User defined method that Returns PyTorch model
+        User defined method that Returns PyTorch model.
+
+        Args:
+            config (Config): The config from the server.
+
+        Returns:
+            nn.Module: The client model.
+
+        Raises:
+            NotImplementedError: To be defined in child class.
         """
         raise NotImplementedError
 
     def update_after_train(self, local_steps: int, loss_dict: Dict[str, float]) -> None:
         """
-        Called after training with the number of local_steps performed over the FL round and
+        Hook method Called after training with the number of local_steps performed over the FL round and
```

Comment:
lowercase called 🙂

Reply:
It might be useful to provide an example here of where this might be used. I believe this is used by SCAFFOLD and FedProx